includes/admin/class-wcmypa-admin.php

```diff
@@ -113,6 +118,91 @@ public function __construct()
         add_action("woocommerce_process_product_meta", [$this, "productOptionsFieldSave"]);
     }
 
+    /**
+     * @throws \Exception
+     */
+    public function restrictManagePosts(): void
+    {
+        global $typenow;
+
+        if (in_array($typenow, wc_get_order_types('order-meta-boxes'))) {
+            $this->deliveryDayFilters();
+        }
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function deliveryDayFilters(): void
+    {
+        if (apply_filters('deliveryDayFilter', true)) {
+            $this->deliveryDayFilter();
+        }
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function deliveryDayFilter(): void
+    {
+        if (is_admin() && ! empty($_GET['post_type']) && $_GET['post_type'] == 'shop_order') {
+            $selected = (isset($_GET['deliveryDate'])
+                ? sanitize_text_field($_GET['deliveryDate'])
+                : false);
+            ?>
+
+            <select name="deliveryDate">
+                <option value=""><?php _e('all_delivery_days', 'woocommerce-myparcel'); ?></option>
+                <?php
+                $carrierName       = WCMYPA_Settings::SETTINGS_POSTNL;
+                $deliveryDayWindow = (int) WCMYPA()->setting_collection->getByName(
+                    $carrierName . "_" . WCMYPA_Settings::SETTING_CARRIER_DELIVERY_DAYS_WINDOW
+                );
+
+                foreach (range(1, $deliveryDayWindow) as $number) {
+                    $date    = date('Y-m-d', strtotime('now' . '+' . $number . 'days'));
+                    $dayDate = wc_format_datetime(new WC_DateTime($date), 'D d-m');
```

Comment:
noem dit `dateString` ofzo, dan is het duidelijker dat dit de tekst is die getoond gaat worden

Reply:
done